src/components/Whiteboard/ElementBehaviors/Resizable/ResizeHandle.tsx

```diff
@@ -16,102 +16,150 @@
 
 import { Dispatch, RefObject, useCallback, useRef } from 'react';
 import { DraggableCore, DraggableData, DraggableEvent } from 'react-draggable';
+import { ElementKind } from '../../../../state';
 import { useSvgCanvasContext } from '../../SvgCanvas';
-import { ResizeHandlePosition } from './types';
-
-export function calculateResizeHandlePosition(
-  position: ResizeHandlePosition,
-  containerWidth: number,
-  containerHeight: number,
-  scale: number,
-): {
+import {
+  LineElementResizeHandlePosition,
+  PolylineAndShapeElementsResizeHandlePosition,
+  ResizeHandlePosition,
+} from './types';
+
+export type BaseParams = {
+  scale: number;
+};
+
+export type LineElementParams = BaseParams & {
+  elementKind: 'line';
+  handlePosition: LineElementResizeHandlePosition;
+  handlePositionX?: number;
+  handlePositionY?: number;
+};
```

Comment:
`handlePositionX` and `handlePositionY` should not be optional, they are defined for lines